typing fixes and improvements

dlax opened the pull request:
A few commits to make mypy happy in the code base.

dlax committed:
Add type parameters to Dict or List

dlax committed:
Use dict and list from builtins in type hints

Since we target Python 3.9, typing.Dict and typing.List are deprecated
and builtins can be used instead.

Richardk2n commented:
Thank you.
The reason for `List` and `Dict` was because you are not able to use the builtins in older python versions.
Do you happen to know, since when this is possible? (To make sure we still support the versions we claim to support)

Did you figure out, why the call overload is complaining?
I never did. It looks fine to me.

dlax committed:
Type windows_flag to fix 'call-overload' mypy error

dlax commented:
> The reason for `List` and `Dict` was because you are not able to use the builtins in older python versions. Do you happen to know, since when this is possible? (To make sure we still support the versions we claim to support)

This is possible since Python 3.9, so fine for us.

> Did you figure out, why the call overload is complaining? I never did.

This is because of the `**windows_flag`; but in fact, I found a better way to fix those, see last commit.

Richardk2n commented:
Ok good.

Can you elaborate a bit?
I am fine with this solution if it works, but am not quite sure why it does.
Also, the `#type ignore` in that line should be removable now?
The tests also have these, so maybe use the typed dict there as well.

dlax committed:
Add the error code about windows_flag's type ignore

This 'type ignore' is because mypy will raise an error on Linux as the
subprocess module does not have a CREATE_NO_WINDOW attribute defined.

dlax committed:
Re-use windows_flag in test

dlax commented:
> Can you elaborate a bit? I am fine with this solution if it works, but am not quite sure why it does.

The previous type hint for `windows_flag: dict[str, int]` resulted in a `subprocess.run(**windows_flag: int)` call (from typing pov) but the function does not accept arbitraty `**kwargs: int`. By changing to a `TypedDict`, we can keep the `**windows_flag` syntax but now the call is `subprocess.run(creationflags: int)`, which is allowed.

> Also, the `#type ignore` in that line should be removable now?

No, it's still needed because, on linux, subprocess has no `CREATE_NO_WINDOW` attribute; I added an error code for that.

> The tests also have these, so maybe use the typed dict there as well.

I changed the test to reuse the value from `plugin` module.

Richardk2n commented:
Ok, thx for the explanation.
Can you please add a comment to the line about the ignoring being necessary even so this never can produce an error?

It seems like you did not run the black formatter.

dlax committed:
Add a comment about 'type: ignore' for WindowsFlag

dlax commented:
Done

dlax commented:
Fixed the flake8 error, sorry about that.

Richardk2n commented:
line 73 still is too long?

dlax commented:
> line 73 still is too long?

fixed...

Richardk2n commented:
Thank you for your contribution.
